pkg/roachprod/install/cluster_synced.go

```diff
@@ -119,6 +119,14 @@ func NewSyncedCluster(
 		return nil, err
 	}
 	c.Nodes = nodes
+
+	if c.ClusterSettings.secureFlagsOpt != nil {
```

Comment:
Nit: for brevity, could fold the nullity check into the method.

Reply:
As this is an interface, this method is bound to have multiple implementation, so this check being implemented multiple times vs. the single call. It feels safer to implement it there once and for all.